components/ConfirmMessage.js

```diff
@@ -0,0 +1,109 @@
+import styled from "styled-components";
+import { useEffect, useState } from "react";
+
+const StyledBackground = styled.div`
+  background-color: black;
+  opacity: 0.5;
+  position: fixed;
+  inset: 0;
+  z-index: 2;
+`;
+const StyledPopUpMessage = styled.div`
+  position: fixed;
+  right: calc(50% - 175.25px);
+  top: 13rem;
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+  background-color: white;
+  padding: 1.5rem;
+  box-shadow: 0 0 5px 0px;
+  border-radius: 10px;
+  transform: ${({ $animation }) => ($animation ? "scale(1)" : "scale(0)")};
+  transition: transform 0.3s;
+  z-index: 3;
+`;
+
+const StyledParagraph = styled.p`
+  padding: 1rem 2rem;
+  text-align: center;
+`;
+
+const StyledWrapper = styled.div`
+  display: flex;
+  width: 100%;
+  justify-content: space-evenly;
+  align-items: center;
+  padding: 1rem;
+`;
+const StyledButton = styled.button`
+  color: white;
+  background-color: ${({ $color }) => $color};
+  border: none;
+  border-radius: 5px;
+  padding: 1rem;
+`;
+
+export default function ConfirmMessage({
+  toggleMessage,
+  itemId,
+  itemText,
+  confirmFunction,
+  children,
+  cancelButtonText,
+  confirmButtonText,
+  cancelButtonColor,
+  confirmButtonColor,
+}) {
+  // for animation-state, useEffect and timeeOuts: animation needs to be triggered after on-mount and before dis-mount of component in order to work with conditional rendering
+  //
+  const [animation, setAnimation] = useState(false);
+
+  useEffect(() => {
+    setAnimation(true);
+    return () => {};
+  }, []);
+
+  function handleCancel(itemId) {
+    setAnimation(false);
+    setTimeout(() => toggleMessage(itemId), 300);
```

Comment:
Thanks for your review! The renaming is definitely much better. 

About the last suggestion. I didn't put the `button` inside the `ListItem` because clicking the `button` would also trigger the `detailView` to open which i would like to avoid. --> I changed the `key` to be on the `wrapper` instead!